tests/unit/alloc_test.cc

```diff
@@ -175,4 +175,23 @@ SEASTAR_TEST_CASE(test_foreign_function_use_glibc_malloc) {
     test_allocation_function([]() { return aligned_alloc(4, 1024); });
     return make_ready_future<>();
 }
+
+// So the compiler won't optimize the call to realloc(nullptr, size)
+// and call malloc directly.
+void* test_nullptr = nullptr;
+
+SEASTAR_TEST_CASE(test_realloc_nullptr) {
+    auto p0 = realloc(test_nullptr, 8);
+    BOOST_REQUIRE(p0 != nullptr);
+    BOOST_REQUIRE_EQUAL(realloc(p0, 0), nullptr);
```

Comment:
Nice. This is an important point that's easily missed - realloc(p0, 0) - when p0!=0, should do free(p). So good you tested it as well.